scheduler/controllers/clustercollectorsgroup/resource_config.go

```diff
@@ -27,37 +38,45 @@ const (
 	memoryLimitAboveRequestFactor = 1.25
 )
 
-// process the memory settings from odigos config and return the memory settings for the collectors group.
+// process the resources settings from odigos config and return the resources settings for the collectors group.
 // apply any defaulting and calculations here.
-func getMemorySettings(odigosConfig *common.OdigosConfiguration) *odigosv1.CollectorsGroupMemorySettings {
-	memoryRequestMiB := defaultRequestMemoryMiB
-	if odigosConfig.CollectorGateway != nil && odigosConfig.CollectorGateway.RequestMemoryMiB > 0 {
-		memoryRequestMiB = odigosConfig.CollectorGateway.RequestMemoryMiB
-	}
-
-	memoryLimitMiB := int(float64(memoryRequestMiB) * memoryLimitAboveRequestFactor)
+func getGatewayResourceSettings(odigosConfig *common.OdigosConfiguration) *odigosv1.CollectorsGroupResourcesSettings {
+	gatewayConfig := odigosConfig.CollectorGateway
 
+	gatewayMinReplicas := getOrDefault(gatewayConfig, gatewayConfig.MinReplicas, MinReplicasDefault)
```

Comment:
`gatewayConfig.MinReplicas` need `nil` check on `gatewayConfig` before dereferencing it